src/service/db/alerts/destinations.rs

```diff
@@ -87,28 +141,28 @@ pub async fn watch() -> Result<(), anyhow::Error> {
         };
         match ev {
             db::Event::Put(ev) => {
-                let item_key = ev.key.strip_prefix(key).unwrap();
-                let item_value: Destination = if config::get_config().common.meta_store_external {
-                    match db::get(&ev.key).await {
-                        Ok(val) => match json::from_slice(&val) {
-                            Ok(val) => val,
-                            Err(e) => {
-                                log::error!("Error getting value: {}", e);
-                                continue;
-                            }
-                        },
-                        Err(e) => {
-                            log::error!("Error getting value: {}", e);
-                            continue;
-                        }
+                let item_key = ev.key.strip_prefix(DESTINATION_WATCHER_PREFIX).unwrap();
+                let keys = item_key.split('/').collect::<Vec<_>>();
+                if keys.len() != 2 {
+                    log::error!("Error event key not formatted correctly. Should be org_id/name, but got {}", item_key);
+                    continue;
+                }
```

Comment:
Nitpick: 
It might be nice to factor key generation and parsing into their own helper methods.